Source/Orts.Simulation/Simulation/RollingStocks/SubSystems/CruiseControl.cs

```diff
@@ -0,0 +1,2277 @@
+                            {
+                                if (Locomotive.DynamicBrakePercent > 0)
+                                {
+                                    if (controllerVolts < 0)
+                                    {
+                                        float step = 100 / DynamicBrakeFullRangeDecreaseTimeSeconds;
+                                        step *= elapsedClockSeconds;
+                                        controllerVolts += step;
+                                    }
+                                }
+                            }
+                        }
+                    }
+
+                    if ((AbsWheelSpeedMpS > SafeSpeedForAutomaticOperationMpS || SpeedSelMode == SpeedSelectorMode.Start || SpeedRegulatorOptions.Contains("startfromzero")) && (SpeedSelMode != SpeedSelectorMode.Neutral && SpeedSelMode != SpeedSelectorMode.Parking))
+                    {
+                        float delta = 0;
+                        if (!RestrictedSpeedActive)
+                            delta = SelectedSpeedMpS - AbsWheelSpeedMpS;
+                        else
+                            delta = CurrentSelectedSpeedMpS - AbsWheelSpeedMpS;
+                        float coeff = 1;
+                        float speed = SpeedIsMph ? MpS.ToMpH(Locomotive.WheelSpeedMpS) : MpS.ToKpH(Locomotive.WheelSpeedMpS);
+                        if (speed > 100)
+                        {
+                            coeff = (speed / 100) * 1.2f;
+                        }
+                        else
+                        {
+                            coeff = 1;
+                        }
+                        float tempAccDemand = AccelerationDemandMpSS;
+                        AccelerationDemandMpSS = (float)Math.Sqrt((StartReducingSpeedDelta) * coeff * (delta));
+                        if (float.IsNaN(AccelerationDemandMpSS))
+                        {
+                            AccelerationDemandMpSS = tempAccDemand;
+                        }
+                        if (delta > 0.0f && Locomotive.DynamicBrakePercent < 1)
+                        {
+                            if (Locomotive.DynamicBrakePercent > 0)
+                            {
+                                float step = 100 / DynamicBrakeFullRangeDecreaseTimeSeconds;
+                                step *= elapsedClockSeconds;
+                                controllerVolts += step;
+                            }
+                            if (Locomotive.DynamicBrakePercent < 1 && Locomotive.DynamicBrake)
+                            {
+                                Locomotive.SetDynamicBrakePercent(0);
+                                Locomotive.DynamicBrakeChangeActiveState(false);
+                            }
+                            relativeAcceleration = (float)Math.Sqrt(AccelerationRampMaxMpSSS * delta);
+                        }
+                        else // start braking
```

Comment:
It seems that this is very similar to a previous block. Is there a duplication that can be avoided?